README.md

```diff
@@ -705,3 +705,132 @@ python runqemu.py --config=NONE --cache=/path/to/cache_dir \
 Older versions of `runqemu.py` would examine the `--image-name` or
 `--image-file` value to determine if the image is an EL6 image, but this is now
 deprecated. You must specify `--ssh-el6` if you need it.
+
+### Container testing
+
+Integration tests can be run using containers.  `tox` will start one or
+more containers for the managed nodes using `podman` and the Ansible `podman` connection
+plugin.  There are some test envs that can be used to run these
+tests:
+* `container-ansible-2.9` - tests against Ansible 2.9
+  * also uses jinja 2.7 if supported by the python used
+* `container-ansible-core-2.x` - tests against ansible-core 2.x
+
+These tests run in one of two modes, depending on which of the following
+arguments you provide.  Note that you must use `--` on the command line after
+the `-e container-ansible-2.9` or `-e container-ansible-core-2.x` so that `tox` will not attempt to
+interpret these as `tox` arguments:
+```
+tox -e container-ansible-core-2.15 -- --image-name fedora-36 ...
+```
+You must provide `--image-name`.
+
+* `--image-name` - assuming you have a config file (`--config`) that maps the
+  given image name to a container registry image and optional setup, you can just specify an
+  image name like `--image-name fedora-38` and the script will pull the
+  latest image for Fedora 38.  In the config file you can specify
+  additional setup steps to be run e.g. setting up additional dnf/yum repos.
+  The corresponding environment variable is `CONTAINER_IMAGE_NAME`.
+* `--config` - default `$HOME/.config/linux-system-roles.json` - this is the
+  full path to a config file that lists the image names, the source or compose,
+  and additional setup steps.  The corresponding environment variable is
+  `CONTAINER_CONFIG`.
+* `--erase-old-snapshot` - If given, erase the current snapshot.  If not specified
+  `tox` will use the current snapshot if it is less than
+  `CONTAINER_AGE` hours old (default: 24)
+* `--parallel N` - N is an integer which is the number of containers to run in
+  parallel.  See below.  The default is `0` which means playbooks will be run
+  sequentially.
+* `--log-dir PATH` - When using `--parallel` - for each test playbook named `TESTNAME.yml`
+  the log will be written to a file named `TESTNAME.log` in the `PATH` directory.
+  The default is the directory where `TESTNAME.yml` is found.
```

Comment:
May I add the description for the new options `--extra-rpm` and `--extra-skip-tag` here?

- --extra-rpm <RPM>: specifies an additional rpm package to install
      in the container. Default to none. E.g.,
      --extra-rpm diffutils --extra-rpm sudo
- --extra-skip-tag <TAG>: specifies a tag to skip. Default to none.
      E.g., --extra-skip-tag "tests::use_selinux_role"